The switch cmake toolchain does not support main from static library

The following cmake project fails to build when using the switch cmake toolchain:
**CMakeLists.txt**
```cmake
cmake_minimum_required(VERSION 3.8)
project(main_from_lib)

add_library(staticlib static.c)

add_executable(executable exe.c)
target_link_libraries(executable PRIVATE staticlib)
```
**static.c**
```c
int some_func(void);
int main() {
	return some_func();
}
```
**exe.c**
```c
int some_func(void) {
	return 0;
}
```
Building with switch-cmake with the current `devkitpro/devkita64` docker images gives the following output (and error):
```
/opt/devkitpro/portlibs/switch/bin/aarch64-none-elf-cmake -S ../nxsamp/ -B.  && make
-- The C compiler identification is GNU 11.1.0
-- The CXX compiler identification is GNU 11.1.0
-- Check for working C compiler: /opt/devkitpro/devkitA64/bin/aarch64-none-elf-gcc
-- Check for working C compiler: /opt/devkitpro/devkitA64/bin/aarch64-none-elf-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/devkitpro/devkitA64/bin/aarch64-none-elf-g++
-- Check for working CXX compiler: /opt/devkitpro/devkitA64/bin/aarch64-none-elf-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /conan/ff
Scanning dependencies of target staticlib
[ 25%] Building C object CMakeFiles/staticlib.dir/static.c.obj
[ 50%] Linking C static library libstaticlib.a
[ 50%] Built target staticlib
Scanning dependencies of target executable
[ 75%] Building C object CMakeFiles/executable.dir/exe.c.obj
[100%] Linking C executable executable
/opt/devkitpro/devkitA64/bin/../lib/gcc/aarch64-none-elf/11.1.0/../../../../aarch64-none-elf/bin/ld: /opt/devkitpro/libnx/lib/libnx.a(switch_crt0.o): in function `_start':
/home/fincs/pacman-packages/libnx/src/libnx-4.1.1/nx/source/runtime/switch_crt0.s:70: undefined reference to `main'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/executable.dir/build.make:85: executable] Error 1
make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/executable.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

This can be fixed by adding `-lnx` twice to the linker flags: once before the user object and libraries and once at the end.

Applying the following patch to `${DEVKITPRO}/cmake/Switch.cmake` fixed the error for me:
```diff
--- /opt/devkitpro/cmake/Switch.cmake	2021-06-14 10:23:20.000000000 +0000
+++ ../Switch.cmake	2021-06-22 15:13:09.613154005 +0000
@@ -12,7 +12,7 @@
 set(CMAKE_CXX_FLAGS_INIT "${NX_COMMON_FLAGS}")
 set(CMAKE_ASM_FLAGS_INIT "${NX_COMMON_FLAGS}")
 
-set(CMAKE_EXE_LINKER_FLAGS_INIT "${NX_ARCH_SETTINGS} ${NX_LIB_DIRS} -fPIE -specs=${DEVKITPRO}/libnx/switch.specs")
+set(CMAKE_EXE_LINKER_FLAGS_INIT "${NX_ARCH_SETTINGS} ${NX_LIB_DIRS} -fPIE -specs=${DEVKITPRO}/libnx/switch.specs -lnx")
 
 set(CMAKE_FIND_ROOT_PATH
 	${CMAKE_FIND_ROOT_PATH}
```

--- a/cmake/switch/PKGBUILD
+++ b/cmake/switch/PKGBUILD
@@ -1,6 +1,6 @@
 # Maintainer: WinterMute <davem@devkitpro.org>
 pkgname=switch-cmake
-pkgver=1.1.0
+pkgver=1.1.1
 pkgrel=1
 pkgdesc="switch cmake wrapper"
 arch=('any')
@@ -15,7 +15,7 @@ groups=('switch-dev')
 options=('!strip')
 sha256sums=(
   '8d19a809aa1db2bb246890398f37c35254cd7a8f46a44dd312c92302352f5927' # aarch64-none-elf-cmake
-  '73ffcbab23b5ef36257095ae1d37a4610127a70a264512d87bd3ecc00c66e1f4' # Switch.cmake
+  '29c31e988e234c83cd88b32a197a0c76306bf983600f27b41ff271cb316a4d96' # Switch.cmake
 )
 
 package() {

--- a/cmake/switch/Switch.cmake
+++ b/cmake/switch/Switch.cmake
@@ -86,11 +86,16 @@ function(nx_generate_nacp outfile)
 	)
 endfunction()
 
-function(nx_create_nro prefix)
+function(nx_create_nro target)
 	cmake_parse_arguments(ELF2NRO "NOICON;NONACP" "ICON;NACP;ROMFS" "" ${ARGN})
 
-	set(ELF2NRO_ARGS ${prefix} "${prefix}.nro")
-	set(ELF2NRO_DEPS ${prefix})
+	get_target_property(TARGET_OUTPUT_NAME ${target} OUTPUT_NAME)
+	if(NOT TARGET_OUTPUT_NAME)
+		set(TARGET_OUTPUT_NAME "${target}")
+	endif()
+
+	set(ELF2NRO_ARGS "$<TARGET_FILE:${target}>" "${TARGET_OUTPUT_NAME}.nro")
+	set(ELF2NRO_DEPS ${target})
 
 	if (DEFINED ELF2NRO_ICON AND ELF2NRO_NOICON)
 		message(FATAL_ERROR "nx_create_nro: cannot specify ICON and NOICON at the same time")
@@ -105,7 +110,7 @@ function(nx_create_nro prefix)
 	endif()
 
 	if (NOT DEFINED ELF2NRO_NACP AND NOT ELF2NRO_NONACP)
-		set(ELF2NRO_NACP "${prefix}.default.nacp")
+		set(ELF2NRO_NACP "${target}.default.nacp")
 		nx_generate_nacp(${ELF2NRO_NACP})
 	endif()
 
@@ -139,15 +144,15 @@ function(nx_create_nro prefix)
 	endif()
 
 	add_custom_command(
-		OUTPUT "${prefix}.nro"
+		OUTPUT "${TARGET_OUTPUT_NAME}.nro"
 		COMMAND "${NX_ELF2NRO_EXE}" ${ELF2NRO_ARGS}
 		DEPENDS ${ELF2NRO_DEPS}
 		VERBATIM
 	)
 
 	add_custom_target(
-		"${prefix}_nro" ALL
-		DEPENDS "${prefix}.nro"
+		"${target}_nro" ALL
+		DEPENDS "${TARGET_OUTPUT_NAME}.nro"
 	)
 endfunction()
 

